### DISCUSSION TITLE
[@types/mapbox__vector-tile] Enhancing Flexibility in VectorTileFeature: Allowing Customization of the properties Type

### DISCUSSION THREAD

**Discussion by anne030303** [DefinitelyTyped/DefinitelyTyped#67867] [Issues with a @types package] [answered]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a69c7b86c69e53fc389687e766c0e2aea2337325/types/mapbox__vector-tile/index.d.ts#L15

In the current implementation of VectorTileFeature, the type of the properties attribute is hardcoded as { [_: string]: string | number | boolean }. This design might restrict users in certain scenarios where they want to precisely define the types of properties.

To enhance the flexibility of VectorTileFeature, it is suggested to change the type of properties to a generic T, allowing users to freely define the types of properties. Additionally, the toGeoJSON method has been proposed with the following signature. The modified code would be as follows:

```
import Pbf = require("pbf");
import { Feature, Point } from "geojson";

export class VectorTile<T> {
    constructor(pbf: Pbf);
    layers: { [_: string]: VectorTileLayer<T> };
}

export class VectorTileFeature<T> {
    static types: ["Unknown", "Point", "LineString", "Polygon"];
    extent: number;
    type: 0 | 1 | 2 | 3;
    id: number;
    properties: T;
    loadGeometry(): Point[][];
    toGeoJSON(x: number, y: number, z: number): Feature<Point, T>;
    bbox?(): [number, number, number, number];
}

export class VectorTileLayer<T> {
    constructor(pbf: Pbf);
    version?: number;
    name: string;
    extent: number;
    length: number;
    feature(featureIndex: number): VectorTileFeature<T>;
}

```

[a comment by an automated account is not shown]

**Comment by HarelM** [comment, marked as answer]
You can override it: `type MyFeature = VectorTileFeature & { properties: { onlyThis: string } }`
